### ISSUE TITLE
OBD-II ignition check power management mode doesn't work if DEBUG=0

### ISSUE THREAD

**Issue by zacnelson** [openxc/vi-firmware#318]
I've tested the 3 released firmware types from the releases page: https://github.com/openxc/vi-firmware/releases/tag/v7.0.0.

I'm only able to get USB and Bluetooth data from the "emulator" firmware type. I tried compiling the firmware on my local machine (in Cygwin) and also got the same result when using the firmware from the `fab reference translated_obd2 build` command. I did notice though, that my local compiled firmware versions are about 40kB smaller than those on the releases folder.

If I have a valid `signals.cpp` file generated with the python tools, then I am able to get data through via BT and USB successfully. I do, however, get the following output errors which don't seem to be an actual error:

````
$ openxc-control version --format json
Device is running version 7.0.0 (type-3)
Can't read from data source -- stopping: ("USB device couldn't be read", USBError(None, 'libusb0-dll:err [_usb_reap_async] timeout error\n'))
Can't read logs from data source -- stopping: ("USB device couldn't be read", USBError(None, 'libusb0-dll:err [_usb_reap_async] timeout error\n'))
````

I'm testing with the Fordboard shipped with the stickers on the outside.

**Comment by peplin** [comment]
Thanks for the report, Zac. I was able to replicate this bug - the firmware boots up to check if the ignition is running, and it does sent a couple of messages over CAN, but it goes to sleep before waiting for the response. It's odd that the CAN activity light stays red with these hex files - it's sending out messages, which should cause it to turn blue. The reason you can't detect it over USB is that the firmware doens't fire up the output interface unless it's decided that CAN is awake, and it will actually feed them data.

I tried a build from `fab reference translated_obd2 build` on the `master branch`, and it worked OK - the interface light goes blue immediately on wakeup when it sends the ignition check messages. I wonder if the release build is somehow bad.

I re-built the translated OBD-II firmware for a reference VI running the bootloader, and that seems to work fine:

https://dl.dropboxusercontent.com/u/980077/vi-firmware-translated_obd2-FORDBOARD-efa2b4ead.bin

If you can confirm that also works, I'll see about doing a re-build of the 7.0 release ZIP.

**Comment by peplin** [comment]
After more testing, I think the root problem is that the OBD-II ignition check power mode doesn't work if DEBUG=0. I am now able to reliable have it work and not work if I flash a DEBUG and non-DEBUG build to bare metal. That definitely gives me something to go on, although I can't think of what might be causing it to fail off the top of my head. The debug code adds logging which tends to slow the code down in places, so it's possible we just need a little delay for non-DEBUG builds so it doesn't shut off too soon.

**Comment by peplin** [comment]
This should be fixed with the new [v7.0.1](https://github.com/openxc/vi-firmware/releases/tag/v7.0.1) release, as of commit 9fc8efb0 - thanks for the bug report @zacnelson, that really helped me track it down.
